### ISSUE TITLE
Support arbitrary task inline fields

### ISSUE THREAD

**Issue by stoilis** [sytone/obsidian-queryallthethings#11]
qatt currently supports querying the standard task fields, like dueDate etc.

However, why does it have to be explicit only for the predefined fields?
Can you open it up to any field that uses the standard [field:: value] format?

qatt alredy makes the fields available for [dataview pages](https://sytone.github.io/obsidian-queryallthethings/examples-tutorials/dataview-fields). But the scope of those is the entire page.

For my use-case, I wanted to query based specifically on the "length" field, which is not very standardised.

**Comment by sytone** [comment]
For tasks yes the parsing can be opened up. I will look at that.

For your second part of the comment is the length field in the yaml front matter and part of the page or part of something else on the page.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Ioannis Stoilis ***@***.***>
Sent: Tuesday, January 30, 2024 12:00:19 PM
To: sytone/obsidian-queryallthethings ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [sytone/obsidian-queryallthethings] Support arbitrary task inline fields (Issue #11)


qatt currently supports querying the standard task fields, like dueDate etc.

However, why does it have to be explicit only for the predefined fields?
Can you open it up to any field that uses the standard [field:: value] format?

qatt alredy makes the fields available for dataview pages<https://sytone.github.io/obsidian-queryallthethings/examples-tutorials/dataview-fields>. But the scope of those is the entire page.

For my use-case, I wanted to query based specifically on the "length" field, which is not very standardised.

**Comment by stoilis** [comment]
The length is an inline field in the task itself. For example:
- [ ] something [length:: 5m]

**Comment by sytone** [comment]
ok, so this related to the request to allow additional fields to be added to the tasks table by parsing any [name:: value] found in the task string.

**Comment by stoilis** [comment]
Exactly 👍

**Comment by sytone** [comment]
ok, ill look at updating the parsing logic. This will impact the `obsidian_markdown_lists` and `obsidian_markdown_tasks` initally, I am no longer using the dataview tables so will have to spend more time to see how to handle that.

**Comment by sytone** [comment]
Did a quick look to see how complex the parsing would be, I now can parse additional properties from the task string. It is hard coded as a test but seems to work. Need to think about plumbing to allow users to configure this.

I assume it is in the format of `prefix:: value` or `[prefix:: value]` at the moment if there is a space in the value it will not handle it as I need to search for a closing bracket but the basics are in place.

```markdown
- [ ] This is a test [length:: 5m]
- [ ] This is a test [length:: 10m]
```

```qatt
query: |
  select top 3 task, page, content, text, prop, length, line, cleanTask
  from obsidian_markdown_tasks
  where content like '%This is a test%'
template: |
  {{#each result}}
  {{stringify this}}
  {{/each}}
```

Output of codeblock:
```text
{
"task": " ",
"page": "1 Journal/2024/2024-01/2024-W05.md",
"content": "- [ ] This is a test [length:: 5m]",
"text": "This is a test [length:: 5m]",
"length": "5m",
"line": 32,
"cleanTask": "- [ ] This is a test"
}
{
"task": " ",
"page": "1 Journal/2024/2024-01/2024-W05.md",
"content": "- [ ] This is a test [length:: 10m]",
"text": "This is a test [length:: 10m]",
"length": "10m",
"line": 33,
"cleanTask": "- [ ] This is a test"
}
```

**Comment by sytone** [comment]
Ok, fixed the parsing if encased in square brackets to allow spaces be part of the value.

```markdown
- [ ] This is a test [length:: 5m]
- [ ] This is a test length:: 10m
- [ ] This is a test [length:: 5 minutes]
- [ ] This is a test [length:: 10 minutes]
```

```qatt
query: |
  select top 5 task, page, content, text, prop, length, line, cleanTask
  from obsidian_markdown_tasks
  where content like '%This is a test%'
template: |
  {{#each result}}
  {{stringify this}}
  {{/each}}
```

Output of codeblock:
```text
{
"task": " ",
"page": "1 Journal/2024/2024-01/2024-W05.md",
"content": "- [ ] This is a test [length:: 5m]",
"text": "This is a test [length:: 5m]",
"length": "5m",
"line": 32,
"cleanTask": "- [ ] This is a test"
}
{
"task": " ",
"page": "1 Journal/2024/2024-01/2024-W05.md",
"content": "- [ ] This is a test length:: 10m",
"text": "This is a test length:: 10m",
"length": "10m",
"line": 33,
"cleanTask": "- [ ] This is a test"
}
{
"task": " ",
"page": "1 Journal/2024/2024-01/2024-W05.md",
"content": "- [ ] This is a test [length:: 5 minutes]",
"text": "This is a test [length:: 5 minutes]",
"length": "5 minutes",
"line": 34,
"cleanTask": "- [ ] This is a test"
}
{
"task": " ",
"page": "1 Journal/2024/2024-01/2024-W05.md",
"content": "- [ ] This is a test [length:: 10 minutes]",
"text": "This is a test [length:: 10 minutes]",
"length": "10 minutes",
"line": 35,
"cleanTask": "- [ ] This is a test"
}
```

**Comment by sytone** [comment]
Will look at how to handle the user driven configuration in the next two days.

**Comment by stoilis** [comment]
Nice. I assume this would expected to work the same with multiple field/value pairs per task, right?

**Comment by sytone** [comment]
Yes. Looking at either user specify g the fields or seeing if there is a 'safe' way to search for key:: value or [key:: value] in the string to automatically pull all of them. Probably have it off by default in case a user has a legitimate double colon in the string and they do not care about this parsing.

**Comment by sytone** [comment]
I have released 0.10.0 which should be avaliable to update in Obsidian now. There is a toggle in settings to enable this feature. Let me know how it goes. It will pull all the inline fields so you do not have to configure anything, just turn on.

You will want to restart to force the in memory cache to be rebuilt. Ill add a command to do that in the future so restart is not needed.

**Comment by sytone** [comment]
@stoilis Were you able to test this in your vault and do you have any feedback or issues. 

If all good I will close the issue up.

**Comment by stoilis** [comment]
Thank you. I will respond by tomorrow.
